PRESIDECMS-2721 data export ability to choose related object fields

pixl8-brayden opened the pull request.

pixl8-brayden committed:
[twgit] Init feature 'feature-PRESIDECMS-2721_data-export-ability-to-choose-related-object-fields'.

pixl8-brayden committed:
PRESIDECMS-2721 data export ability to choose related object fields

pixl8-brayden committed:
PRESIDECMS-2721
- restrict nesting to single level
- add defensive check when process nested export fields

pixl8-brayden committed:
PRESIDECMS-2721 ability to only exclude field when it is nested export fields

pixl8-brayden committed:
PRESIDECMS-2721
- include more options to configure data export fields
- move logic to service

pixl8-brayden committed:
PRESIDECMS-2721 fix failed test

pixl8-brayden committed:
PRESIDECMS-2721 only process nested fields logic when enabled

sebduggan commented on system/handlers/admin/DataManager.cfc:
```diff
@@ -3382,6 +3383,20 @@ component extends="preside.system.base.AdminHandler" {
+		args.expandNestedFields = appEnableExpandExportFields || objEnableExpandExportFields;
+		if ( !args.expandNestedFields ) {
+			for ( var field in args.selectFields ) {
+				writeDump(field);
```
Debug code!

sebduggan commented on system/handlers/admin/DataManager.cfc:
```diff
@@ -3382,6 +3383,20 @@ component extends="preside.system.base.AdminHandler" {
+				}
+			}
+		}
+
```
Not sure we need any of the above logic. In here, if we're using _exportDataAction, shouldn't expandNestedFields always be true?

The custom exports send their select fields direct to the service, which is what causes the issue as they might already be aliased...

pixl8-brayden committed:
PRESIDECMS-2721 always expand nested field when using datamanger ootb data export action

sebduggan committed:
PRESIDECMS-2721 Use double underscore to separate nested field names

sebduggan committed:
PRESIDECMS-2721 Better formatting for nested select options

pixl8-brayden committed:
PRESIDECMS-2721
- remove global data export allow expand config
- fix data export custom template issue
- update expand field property attribute to allow more customisation

sebduggan commented on system/services/dataExport/DataExportService.cfc:
```diff
@@ -699,6 +715,31 @@ component {
+
+			if ( IsBoolean( arguments.propAttributes.dataExportExpandFields ) ) {
+				processed.shouldExpand = arguments.propAttributes.dataExportExpandFields;
+			} else if ( ListLen( arguments.propAttributes.dataExportExpandFields ) > 0 ) {
```
I think this simplifies to:

`} else if ( Len( arguments.propAttributes.dataExportExpandFields ) ) {`

pixl8-brayden committed:
PRESIDECMS-2721 simplify list len logic check
